File: 47주차/h-index.js
function solution(citations) {
  var answer = -1;

  const n = citations.length;
  citations.sort((a, b) => a - b);
  for (let i = citations[n - 1]; i >= 0; i--) {
    // 인용된 가장 큰값부터 하나씩 체크하기.
    let h = i;
    let cnt = 0;
    for (let j = n - 1; j >= 0; j--) {
      if (citations[j] >= h) cnt += 1; // h번이상 인용되었다면 +1.
    }
    if (cnt >= h) answer = Math.max(answer, h); // cnt가 h편 이상이라면, answer update.
  }

  return answer;
}
// h번이상 인용된 것이 h편이상, 나머지가 h번이하 인용. h의 최댓값.
// => citiations 중에서 가장 큰 값부터 확인.

File: 47주차/h-index.cpp
#include <string>
#include <vector>
#include<algorithm>
#include<cmath>
using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    int n = citations.size();
    sort(citations.begin(),citations.end());
    int maxV = citations[n-1];
    for (int i = maxV; i >= 0; i--){
        int h = i;
        int cnt = 0;
        for (int j = n-1; j >= 0;j--){
            if (citations[j]>=h) cnt += 1;
        }
        if (cnt >= h) answer = max(answer,h);
    }
    
    return answer;
}// 플그머 정렬 파트. c++로 풀기